app/components/groups/creating-groups/action.controller.js

```diff
@@ -0,0 +1,88 @@
+;(function () {
+  angular
+    .module("todo")
+    .controller("ActionController", ActionController);
+
+  ActionController.$inject = ["$scope", "CreatingGroupService", "$modal", "GroupStorage"];
+
+  function ActionController($scope, CreatingGroupService, $modal, GroupStorage) {
+    var self = this;
+    self.newTextGroup = newTextGroup;
+    self.newListGroup = newListGroup;
+    self.action = "default";
```

Comment:
what is `action` and `mode`? It's not obvious at all. 
and please use constants for strings